### ISSUE TITLE
Tabs

### ISSUE THREAD

**Issue by skaller** [eao197/so-5-5#18]
Please stop using tabs in source code. Tabs are hang-over from the 1960s when typists saved keystrokes and improved accuracy by use of the Tab key and tab stops. They were also used by computers to control IBM Selectric typewriters to aid printing forms. This was important because one never knew exactly where the printed form would end up on the paper until it arrived from the printer.

All modern editors allow programming the tab key to insert spaces, assisting in alignment and assuring all clients get the same layout. 

Tab characters are a disaster in source code. The code becomes unreadable on some devices such as my console. They're also indistinguishable from spaces in my editor. Error diagnostics cannot correctly refer to column numbers since the tab stops are viewing-device dependent.

I use 2 character indents, which is the best format for books (which have small line widths), however 4 is acceptable. 3 is silly. 8 is way too much. 8 is the default on my console. My editor is fixed to 2. 
Invariant layout is incompatible with use of tabs.

**Comment by eao197** [comment]
> Please stop using tabs in source code.

Just no.

There is infinite flame about "tabs vs spaces" and there are strong arguments on each side. I don't want to step into that flame. Projects developed by our team use and will use tabs instead of spaces. Period. ;)

**Comment by skaller** [comment]
OK, it is your call. 

Though i don't know of any arguments in favour.
The flame wars on this were settled 2 decades ago.

I have no idea how to find where an error is if it says column:line, which clang does, and I can't tell what it means by the column because i can't even see if there's a tab there or not. How do you handle this? What tab stops do you use?

**Comment by eao197** [comment]
I used spaces in the past. I found difficult to use code fragments with spaces in documentation, articles, blog-post and so on. Another reason -- evolution of displays. Two spaces was OK when I worked on 13" CRT display with EGA card. But even on modern 13" notebooks with FullHD 16:9 displays it is inappropriate. And I can't easily change indentation from 2 spaces to 4 spaces. This is not a problem at all with tabs.

Usually I use 3 as tab stop.

> How do you handle this?

In most cases even line number reported by compiler is not enough to find an actual error (especially in complex code with heavy usage of templates). So I don't even look at column reported by clang :)
